### PR TITLE
Add sysfs and pci disk plug methods and disk API update

### PR DESCRIPTION
**Pull request by katlapinka**

### CODE REVIEW

File: test_utils/disk_finder.py [hunk 1]
```diff
@@ -87,6 +87,32 @@ def discover_ssd_devices(block_devices, devices_res):
...
+                                          "grep 'Unit serial number:' | "
+                                          "awk '{print $NF}'").stdout
+            if not serial:
+                serial = TestRun.executor.run(f"udevadm info --query=all --name={dev} | "
+                                              "grep 'ID_SERIAL' | "
+                                              "awk --field-separator '=' '{print $NF}'").stdout
+                if not serial:
+                    TestRun.LOGGER.warning(f"Device {dev} does not have a serial number.")
+                    serial_numbers[dev] = dev
+                    continue
```
[the remark this answers is not in the document]
Comment (katlapinka) [reply test_utils/disk_finder.py]: done

File: storage_devices/disk.py [hunk 2]
```diff
@@ -143,6 +145,10 @@ def umount_all_partitions(self):
             f"Umounting all partitions from: {self.system_path}")
         cmd = f'umount -l {self.system_path}*?'
         output = TestRun.executor.run(cmd)
+        if output.exit_code != 0 and "mountpoint not found" not in output.stderr:
+            TestRun.LOGGER.warning(
+                f"There was a problem while unmounting device {self.system_path} partitions. "
+                f"stderr: {output.stderr}")
```
[the remark this answers is not in the document]
Comment (katlapinka) [reply storage_devices/disk.py]: done

File: storage_devices/disk.py [hunk 3]
```diff
@@ -151,10 +153,117 @@ def remove_partitions(self):
...
+            if self.serial_number not in serial_numbers:
+                return False
+            else:
+                self.device_name = serial_numbers[self.serial_number]
+                self.system_path = f"/dev/{self.device_name}"
+                return True
+        elif self.system_path:
+            output = fs_utils.ls_item(f"{self.system_path}")
+            return fs_utils.parse_ls_output(output)[0] is not None
+        TestRun.exception("Couldn't check if device is detected by the system")
```
Comment (robertbaldyga) [inline storage_devices/disk.py]: There is no longer `exception()` method in `TestRun`. Please just rise exception.

File: storage_devices/disk.py [hunk 4]
```diff
@@ -151,10 +153,117 @@ def remove_partitions(self):
...
+
+    def wait_for_plug_status(self, should_be_visible):
+        if not wait(lambda: should_be_visible == self.is_detected(),
+                    timedelta(minutes=1),
+                    timedelta(seconds=1)):
+            TestRun.exception(f"Timeout occurred while tying to "
```
Comment (robertbaldyga) [inline storage_devices/disk.py]: Ditto.

File: storage_devices/disk.py [hunk 5]
```diff
@@ -151,10 +153,117 @@ def remove_partitions(self):
...
+        if self.is_detected():
+            return
+        self.execute_plug_command()
+        self.wait_for_plug_status(True)
+
+    def unplug(self):
+        if not self.is_detected():
+            return
+        if not self.device_name:
+            TestRun.exception("Couldn't unplug disk without disk id in /dev/.")
```
Comment (robertbaldyga) [inline storage_devices/disk.py]: Ditto.

File: storage_devices/disk.py (hunk 3 above)
Comment (katlapinka) [reply storage_devices/disk.py]: done

File: storage_devices/disk.py (hunk 4 above)
Comment (katlapinka) [reply storage_devices/disk.py]: done

File: storage_devices/disk.py (hunk 5 above)
Comment (katlapinka) [reply storage_devices/disk.py]: done

**Pull request merged**